src/mumble/GlobalShortcut_win.cpp

```diff
@@ -424,6 +424,39 @@ void GlobalShortcutWin::run() {
 	} while (!isInterruptionRequested());
 }
 
+bool GlobalShortcutWin::isRateLimited() {
+	if (throttleThreshold > 0) {
+		throttleCounter++;
+
+		if (throttleCounter < throttleThreshold) {
+			// Only let every throttleThreshold-th event through
+			return true;
+		}
+
+		throttleCounter = 0;
+
+		if (m_msgQueue.size() < GlobalShortcutWin::QUEUE_CAPACITY / 2) {
+			// Queue is half empty, relax threshold
+			throttleThreshold--;
+			if (throttleThreshold == 0) {
+				qInfo() << "GlobalShortcutWin: Throttle threshold disabled";
+			}
+		}
+	}
+
+	return false;
+}
+
+void GlobalShortcutWin::tryPlaceEvent(std::unique_ptr< MsgRaw > &&event) {
+	const bool enqueued = m_msgQueue.try_emplace(std::move(event));
+	if (!enqueued) {
+		// Cap throttle at 8192 to prevent excessive event dropping while still protecting the queue
+		throttleThreshold = std::min(static_cast< unsigned int >(8192), throttleThreshold + 1);
```

Comment:
`8192` is oddly specific - where does that number come from?

Reply:
> given that every increase by 1 halves the number of events that enter the queue.

I don't think it does :thinking: 
If the threshold is `n`, then only every `n-th` event is let through. That's no exponential slowdown - with every increase in `n`, we have to emit one more event in order to pass the barrier.

Also, I just realized, `8192 == 2^13`. That makes the number seem much less random :sweat_smile: 
Maybe writing it as `1 << 13` makes that more obvious. But really, this is just a really minor thing.

Reply:
Right, it does not half the events

Yes, I started with 1024 and was like "Well I can see 1000 events hitting the queue at once... Maybe I should double the limit" and then I doubled until I was happy at 8192